cornell-dti/course-plan
# Slot settings menu

### Summary

This PR implements the Menu that pops up when you click on the settings button in the slot.
![image](https://user-images.githubusercontent.com/55263191/118381848-4e2dc180-b5bd-11eb-9c7d-175a86ce6709.png)

- [x] Renamed some variables to cater to the from the deleteCourseModal from the resetConfirmationModal
- [x] Edited TeleportModal to be even more flexible (custom modal positioning)
- [x] Create SlotMenu 

Remaining TODOs:

- [ ] Implement edit functionality / make override requirement modal

### Test Plan
- Make sure delete functionality works
- Make sure frontend looks good

### Notes
- Had to use Teleport because it's embedded into the requirements bar
